### DISCUSSION TITLE
SAML Login with Zitadel not working

### DISCUSSION THREAD

**Discussion by zmbcgn** [kimai/kimai#5102] [General]
Hi, 

I'm trying to connect my Kimai Docker setup with Zitadel as SAML IdP. Login works upon the point that I am getting redirected to Zitadel and back, but it still doesn't log me into the page itself.

I guess there might be some attribute issue, but I confirmed already that there is a "Roles" Attribute, and the roles are being mapped correct (according to the Kimai configuration)

I didn't want to open an issue for this as this not really a Bug or something. I tried to "copy" the configuration for Keycloak (https://www.kimai.org/documentation/saml-keycloak.html), as this worked already most of the time with other apps. But no success. Also, I have debug enabled, but I don't know where to look for the output of that debug. 

It clearly doesn't go to stdin.

Is there anyone who accomplished this setup?

For sure, I am happy to contribute my configuration, once it is working, to the Kima Documation.

Best regards
phil

ps: https://zitadel.com/ and https://zitadel.com/docs

**Comment by kevinpapst** [comment]
Usually that is an attribute issues and those should be logged to kimai/var/log/prod.log

**Reply by zmbcgn** [reply]
This file doesn't exist in my setup.

root@e2d6e5864025:/opt/kimai/var/log# ls
root@e2d6e5864025:/opt/kimai/var/log#

**Reply by kevinpapst** [reply]
Then check your docker logs after trying the login

**Reply by zmbcgn** [reply]
This is from the stdin:

```
localhost:8001 172.22.0.2 - - [09/Oct/2024:13:48:53 +0200] "GET /auth/saml/login HTTP/1.1" 302 4232 "https://time.maisec.de/de/login" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:131.0) Gecko/20100101 Firefox/131.0"
[2024-10-09T11:48:55.861643+00:00] deprecation.INFO: Deprecated: explode(): Passing null to parameter #2 ($string) of type string is deprecated {"exception":"[object] (ErrorException(code: 0): Deprecated: explode(): Passing null to parameter #2 ($string) of type string is deprecated at /opt/kimai/vendor/robrichards/xmlseclibs/src/XMLSecurityKey.php:325)"} {"channel":"deprecation"}
localhost:8001 172.22.0.2 - - [09/Oct/2024:13:48:55 +0200] "POST /auth/saml/acs HTTP/1.1" 302 940 "https://auth.nettailor.net/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:131.0) Gecko/20100101 Firefox/131.0"
localhost:8001 172.22.0.2 - - [09/Oct/2024:13:48:55 +0200] "GET /en/login HTTP/1.1" 200 2157 "https://auth.nettailor.net/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:131.0) Gecko/20100101 Firefox/131.0"
localhost:8001 127.0.0.1 - - [09/Oct/2024:13:49:05 +0200] "GET / HTTP/1.1" 302 626 "-" "curl/7.88.1"
localhost:8001 127.0.0.1 - - [09/Oct/2024:13:49:26 +0200] "GET / HTTP/1.1" 302 626 "-" "curl/7.88.1"
```

Is this "deprecated" warning, the root cause? Because other than that, I don't see any issue. I also see some warnings that I don't have to use API password, but the android app isn't able to handle token. But this another topic..

**Reply by kevinpapst** [reply]
A deprecation shouldn't be an issue, but who knows.

You can try to add some debug output to these two files:
- https://github.com/kimai/kimai/blob/main/src/Saml/SamlAuthenticator.php
- https://github.com/kimai/kimai/blob/main/src/Saml/SamlProvider.php

**Reply by zmbcgn** [reply]
Ok. I will come back to this, 

Meanwhile I noticed that in the SAML Request the "RelayState" is set to http://time.maisec.de/en/homepage. Which doesn't exists, as this page is only reachable through HTTPs. Do you know where this "RelayState" comes from?

**Comment by zmbcgn** [comment]
Meanwhile, I switched to "authentik", as there is an official documentation for using kimai. But I am having issues with this as well. 

For the reason, that the title is now misleading, I close this discussion.
